src/main.rs

```diff
@@ -936,8 +936,61 @@ fn do_cmd_certify(
                     is_fresh_import: false,
                 });
         }
-        CertifyKind::Delta { from, to } => {
+        CertifyKind::Delta { mut from, to } => {
             let importable = from.git_rev.is_none() && to.git_rev.is_none();
+            let mut who = who;
+
+            // Collapse a non-importable delta audits which have adjacent versions and compatible
+            // criteria.
+            //
+            // We merge an adjacent audit for a prior version with the new audit (updating the new
+            // audit). The later `update_store` call will remove the now-unused prior audit.
+            if !importable && sub_args.collapse_non_importable {
+                let mapper = CriteriaMapper::new(&store.audits.criteria);
+                let certify_criteria = mapper.criteria_from_list(&criteria);
+                for audit in store
+                    .audits
+                    .audits
+                    .get(&package)
+                    .into_iter()
+                    .flatten()
+                    .filter(|a| !a.importable)
+                {
+                    let audit_criteria = mapper.criteria_from_list(&audit.criteria);
+                    if let AuditKind::Delta {
+                        from: audit_from,
+                        to: audit_to,
+                    } = &audit.kind
+                    {
+                        if audit_to == &from && certify_criteria.contains(&audit_criteria) {
+                            // Consume the existing audit's `from`, `who`, and `notes`.
+                            from = audit_from.clone();
+                            let unique_who: Vec<_> = audit
+                                .who
+                                .iter()
+                                .filter(|s| !who.contains(s))
+                                .cloned()
+                                .collect();
+                            // Precede with the `who` from the audit which aren't in this
+                            // ceritification, to retain a temporal ordering of auditors.
+                            // TODO: is this unnecessary?
+                            who.splice(..0, unique_who);
+
+                            // Precede the certification notes with the notes from the audit.
+                            if let Some(audit_notes) = &audit.notes {
+                                match notes {
+                                    None => notes = Some(audit_notes.clone()),
+                                    Some(n) => {
+                                        notes = Some(format!("{}\n{}", audit_notes, n));
+                                    }
+                                }
+                            }
```

Comment:
Yeah, really wasn't sure what the best choice there is. It just felt wrong to ignore the old notes.

Reply:
Would it be better if it did something like `format!("{}\nNotes since {from}:\n{}", audit_notes, n)`?

Reply:
TBH not sure what works better. 

Perhaps in an ideal world, what we'd do is detect that we want to collapse earlier on in the process, before the user has filled out the certification text and written their notes, and include the original notes as the default value for notes at that point so that they can update/edit it then. That might be a pain to change though.

Reply:
Ah yeah, that's a good idea. I think it's probably okay to hold off on that until someone complains. I think it's possible people won't be adding many notes to git versions, but we'll see...